### PR TITLE
docs: 5.0 release changelog

### PR DESCRIPTION
**Pull request by P0lip**
I still need to mention all the changes we made to rulesets and most probably a few other issues/features.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,11 +5,37 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## Unreleased
+## [5.0.0] - 2019-12-xx
+
+### Features
+- Invalid JSON pointers are reported as errors now [json-ref-resolver#140](https://github.com/stoplightio/json-ref-resolver/pull/140) and [json-ref-resolver#147](https://github.com/stoplightio/json-ref-resolver/pull/147)
```
Comment (philsturgeon) [inline CHANGELOG.md]: This is potentially a fix.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,11 +5,37 @@ All notable changes to this project will be documented in this file.
...
+- Implemented ignoreUnsupportedFormats to make it easier to detect unrecognized formats [#678](https://github.com/stoplightio/spectral/issues/678)
+- Rule's Given can be an array now [#799](https://github.com/stoplightio/spectral/pull/799)
+- Casing built-in function is added [#564](https://github.com/stoplightio/spectral/issues/564)
 
 ### Changed
+- Improved error source detection [#685](https://github.com/stoplightio/spectral/pull/685)
 - Built-in OAS rule `openapi-tags` is now recommended [#706](https://github.com/stoplightio/spectral/pull/706)
 - CLI: Default `--fail-severity` is now `warn`, so getting an `info` or a `hint` will not return a exit status code [#706](https://github.com/stoplightio/spectral/pull/706)
+- When was removed from Rule [#585](https://github.com/stoplightio/spectral/issues/585)
+- Rules are recommended by default [#719](https://github.com/stoplightio/spectral/pull/719)
```
Comment (philsturgeon) [inline CHANGELOG.md]:
```suggestion
- BREAKING: Rules are recommended by default [#719](https://github.com/stoplightio/spectral/pull/719)
```

Maybe put this at the top of the section?

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,11 +5,37 @@ All notable changes to this project will be documented in this file.
...
+- Rulesets no longer need to have `rules` placed [#652](https://github.com/stoplightio/spectral/pull/652)
+
+### Fixed
+- Circular remote references with JSON pointers are resolved correctly [json-ref-resolver#141](https://github.com/stoplightio/json-ref-resolver/pull/141)
+- Local root JSON pointers are resolved correctly [json-ref-resolver#146](https://github.com/stoplightio/json-ref-resolver/pull/146) 
+- Unixify glob patterns under Windows [#679](https://github.com/stoplightio/spectral/issues/679)
+- Improved duplicate keys detection [#751](https://github.com/stoplightio/spectral/issues/751)
+- Spectral should be usable in non-secure environments [#846](https://github.com/stoplightio/spectral/pull/846)
```
Comment (philsturgeon) [inline CHANGELOG.md]: Can we say something like "in browsers without secure number generators" or something more specific? Lots of these changelog items feel a little vague and without clicking on the issue i cant quite figure out what they mean.

File: CHANGELOG.md (hunk 2 above)
Comment (P0lip) [reply CHANGELOG.md]: I moved all breaking to the top.

**Comment by P0lip** [comment]
@philsturgeon I believe I didn't miss anything, but if you could double-check, that'd be great.
I want to make sure we don't exclude anyone and anything.

**Comment by marbemac** [comment]
Looks pretty good to me - should we have a basic v4 -> v5 migration guide? How much work would that be?

**Comment by P0lip** [comment]
> Looks pretty good to me - should we have a basic v4 -> v5 migration guide? How much work would that be?

Yup, the plan was to add a migration guide. I'm going to write some.

**Comment by P0lip** [comment]
Added some here https://github.com/stoplightio/spectral/pull/859/commits/8ea215b63931585baa16e85a62db08ef083bca55

**Comment by P0lip** [comment]
Merged in https://github.com/stoplightio/spectral/pull/878

**Pull request closed without merging**